docs/user-guide/progressive-release/extending.md

```diff
@@ -0,0 +1,122 @@
+---
+template: main.html
+---
+
+# Extending progressive release to other deployment environments
+
+The `release` chart can be easily extended to include other deployment environments. Please consider contributing any extensions to Iter8. We briefly describe how to extend the chart for an [Knative](https://knative.dev/docs/) application. 
+
+## Approach
+
+Modify the release chart after forking the [Iter8 project](https://github.com/iter8-tools/iter8). The chart to be extended is in the `charts/release` sub-folder. The file `release.yaml` is the starting point. For each valid environment, the chart contains a set of files defining the resources that should be created.  These may include:
+
+- the application object(s)
+- [routemaps](../routemap.md) for different traffic patterns
+- configmaps used to specify request distribution (blue-green pattern only)
+- a service defining a common entry for requests (if needed)
+
+Note that the file naming helps identify related template files.
+
+## Example (Knative Service)
+
+For example, to implement a blue-green release for Knative services, the following files could be added.
+
+- `_knative-istio.tpl` - describing the application objects should be deployed
+- `_knative-istio.version.ksvc.tpl` - describe the Knative service object that should be deployed for a version
+- `_knative-istio.blue-green.tpl` - identifies any objects that should be deployed to support the blue-green traffic pattern
+- `_knative-istio.blue-green.routemap.tpl` - the routemap definition
+- `_knative-istio.service.tpl` - a supporting external service
+- `_knative.helpers.tpl` - some supporting functions
+
+An implementation of these is [here](https://github.com/iter8-tools/docs/tree/v0.18.11/samples/knative-bg-extension).
+
+Finally, update `release.yaml` to include `knative-istio` as a valid option:
+
+```tpl
+{{- else if eq "knative-istio" .Values.environment }}
+  {{- include "env.knative-istio" . }}
+```
+
+## Extend the controller
+
+The Iter8 controller will need to be extended to give permission to Iter8 to watch Knative service objects. Configure the deployment of the controller to enable this, (re-)install the controller using the following additional options:
+
+```shell
+--set resourceTypes.ksvc.Group=serving.knative.dev \
+--set resourceTypes.ksvc.Version=v1 \
+--set resourceTypes.ksvc.Resource=services \
+--set "resourceTypes.ksvc.conditions[0]=Ready"
+```
+
+## Using the modified chart
+
+Reference the location of the local copy of the chart instead of using the `--repo` and `--version` options. For example assuming the location is `$CHART`, a deployment of 2 versions of the Knative `hello` service with a 30-70 traffic split would be:
```

Comment:
```suggestion
Reference the location of the local copy of the chart instead of using the `--repo` and `--version` options. For example assuming the location is `$CHART`, a deployment of two versions of the Knative `hello` service with a 30-70 traffic split would be:
```